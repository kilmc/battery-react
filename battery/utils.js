// Utilities
export const compose = (...functions) =>
  functions.reduce((f, g) => (...xs) => f(g(...xs)));

export const identity = x => x;

const fs = require('fs-extra');
const path = require('path');

// Get absolute paths of all directories under packages/*
export const getFiles = (dir) => {
  return fs
    .readdirSync(dir)
    .map(file => path.resolve(dir, file))
    .filter(f => fs.lstatSync(path.resolve(f)).isFile());
};