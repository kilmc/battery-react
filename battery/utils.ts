import * as fs from 'fs-extra';
import * as path from 'path';

// Utilities
export const squash = (fns: Array<(x: number) => number>) => (x: number) =>
  fns.reduce((accum, fn) => fn(accum), x);

export const identity = <A>(x: A) => x;

// Get absolute paths of all directories under packages/*
export const getFiles = (dir: string) => {
  return fs
    .readdirSync(dir)
    .map(file => path.resolve(dir, file))
    .filter(f => fs.lstatSync(path.resolve(f)).isFile());
};
