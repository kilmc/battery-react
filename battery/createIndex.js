const path = require('path');
const cc = require('change-case');

import { getFiles } from './utils';

const ATOMS_DIR = path.resolve(__dirname, '../atomic/src/atoms')

export const printImport = (filepath) => {
  const filename = path.basename(filepath, '.js')
  return `import * as ${cc.pascal(filename)} from './src/atoms/${filename}';`;
}

export const createIndex = (arr) => {
  return arr.map(printImport)
}

export const printCompile = (arr) => {
  const compileList = arr.map(x =>  path.basename(x, '.js'))
    .map(x => cc.pascalCase(x))
    .map(x => '...' + x)
    .join(',\n')

  return `export const compiled = compile({${compileList}})`
}