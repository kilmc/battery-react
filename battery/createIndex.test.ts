// eslint-env jest, node
const path = require('path');

import { createIndex, printCompile } from './createIndex';

import { getFiles } from './utils';

const ATOMS_DIR = path.resolve(__dirname, '../atomic/src/atoms');

describe('createIndex', () => {
  it('matches snapshot', () => {
    expect(createIndex(getFiles(ATOMS_DIR))).toMatchSnapshot();
  });
});

describe('printCompile', () => {
  it('matches snapshot', () => {
    expect(printCompile(getFiles(ATOMS_DIR))).toMatchSnapshot();
  });
});
