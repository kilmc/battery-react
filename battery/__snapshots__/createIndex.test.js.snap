exports[`createIndex matches snapshot 1`] = `
Array [
  "import * as BackgroundColor from \'./src/atoms/background-color\';",
  "import * as BackgroundImage from \'./src/atoms/background-image\';",
  "import * as BackgroundPosition from \'./src/atoms/background-position\';",
  "import * as BackgroundRepeat from \'./src/atoms/background-repeat\';",
  "import * as BackgroundSize from \'./src/atoms/background-size\';",
  "import * as BorderColor from \'./src/atoms/border-color\';",
  "import * as BorderRadius from \'./src/atoms/border-radius\';",
  "import * as BorderWidth from \'./src/atoms/border-width\';",
  "import * as Border from \'./src/atoms/border\';",
  "import * as BoxShadow from \'./src/atoms/box-shadow\';",
  "import * as BoxSizing from \'./src/atoms/box-sizing\';",
  "import * as Clear from \'./src/atoms/clear\';",
  "import * as Color from \'./src/atoms/color\';",
  "import * as Cursor from \'./src/atoms/cursor\';",
  "import * as Display from \'./src/atoms/display\';",
  "import * as Fill from \'./src/atoms/fill\';",
  "import * as Flexbox from \'./src/atoms/flexbox\';",
  "import * as Float from \'./src/atoms/float\';",
  "import * as FontFamily from \'./src/atoms/font-family\';",
  "import * as FontSize from \'./src/atoms/font-size\';",
  "import * as FontStyle from \'./src/atoms/font-style\';",
  "import * as FontWeight from \'./src/atoms/font-weight\';",
  "import * as HeightWidth from \'./src/atoms/height-width\';",
  "import * as LetterSpacing from \'./src/atoms/letter-spacing\';",
  "import * as LineHeight from \'./src/atoms/line-height\';",
  "import * as ListStyle from \'./src/atoms/list-style\';",
  "import * as MarginPadding from \'./src/atoms/margin-padding\';",
  "import * as Opacity from \'./src/atoms/opacity\';",
  "import * as Outline from \'./src/atoms/outline\';",
  "import * as Overflow from \'./src/atoms/overflow\';",
  "import * as PointerEvents from \'./src/atoms/pointer-events\';",
  "import * as Position from \'./src/atoms/position\';",
  "import * as Resize from \'./src/atoms/resize\';",
  "import * as Stroke from \'./src/atoms/stroke\';",
  "import * as TextAlign from \'./src/atoms/text-align\';",
  "import * as TextDecoration from \'./src/atoms/text-decoration\';",
  "import * as TextOverflow from \'./src/atoms/text-overflow\';",
  "import * as TextTransform from \'./src/atoms/text-transform\';",
  "import * as VerticalAlign from \'./src/atoms/vertical-align\';",
  "import * as Webkit from \'./src/atoms/webkit\';",
  "import * as WhiteSpace from \'./src/atoms/white-space\';",
  "import * as WordBreak from \'./src/atoms/word-break\';",
  "import * as ZIndex from \'./src/atoms/z-index\';",
]
`;

exports[`printCompile matches snapshot 1`] = `
"export const compiled = compile({...BackgroundColor,
...BackgroundImage,
...BackgroundPosition,
...BackgroundRepeat,
...BackgroundSize,
...BorderColor,
...BorderRadius,
...BorderWidth,
...Border,
...BoxShadow,
...BoxSizing,
...Clear,
...Color,
...Cursor,
...Display,
...Fill,
...Flexbox,
...Float,
...FontFamily,
...FontSize,
...FontStyle,
...FontWeight,
...HeightWidth,
...LetterSpacing,
...LineHeight,
...ListStyle,
...MarginPadding,
...Opacity,
...Outline,
...Overflow,
...PointerEvents,
...Position,
...Resize,
...Stroke,
...TextAlign,
...TextDecoration,
...TextOverflow,
...TextTransform,
...VerticalAlign,
...Webkit,
...WhiteSpace,
...WordBreak,
...ZIndex})"
`;
